TabulateSmarterTestAdminPackage.Common/Validators/ValidatorCollection.cs

```diff
@@ -0,0 +1,55 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+
+namespace TabulateSmarterTestAdminPackage.Common.Validators
+{
+    public class ValidatorCollection
```

Comment:
I think we can clean this up a bit.

First, let's make the abstract Validator class inherit from an interface:

```
    public interface IValidator
    {
        bool IsValid(object value);
        string GetMessage();
    }
```

And then we can make the ValidatorCollection in herit from the same interface.  So essentially you could pass an `IValidator` to someone and they could use it and it could be a single `Validator` or a `ValidatorCollection` and they wouldn't need to know or care.  Also, by inheriting from a collection already we wouldn't need to hold a collection as a private variable.

Use `List<>` will guarantee ordering of how they were added so that shuold work fine.  I think this class could look like this:

```
    public class ValidatorCollection : List<Validator>, IValidator
    {
        private string _message;

        public string GetMessage()
        {
            return _message;
        }

        public bool IsValid(object value)
        {
            var isValid = this.All(x => x.IsValid(value));

            if (!isValid)
            {
                _message = this.Where(x => !x.IsValid(value))
                    .Select(x => x.GetMessage())
                    .Aggregate((i, j) => i + j);
            }
            else
            {
                _message = null;
            }
            

            return isValid;
        }
    }
```